vitest.config.e2e.ts

```diff
@@ -26,6 +26,12 @@ export default defineConfig({
       },
     },
   },
+  ssr: {
+    resolve: {
+      // FIXME: this is a bug in Vitest, resolving externalized modules should use externalConditions
+      conditions: ['node', 'module-sync'],
+    },
+  },
```

Comment:
@hi-ogawa It seems Vitest uses `resolve.conditions` for both externalized files and no-externalized files. Is it possible to configure Vitest to only use `module-sync` for externalized files? (since `module-sync` should not be used for no-externalized files)

Reply:
Yes, currently Vitest (technically due to vite-node) cannot use `resolve.externalConditions` for server external resolution and only uses `resolve.conditions` for everything. I haven't followed yet but @sheremet-va might know better what will change in https://github.com/vitest-dev/vitest/pull/8208.

> Is it possible to configure Vitest to only use module-sync for externalized files?

What's the desired goal for this diff specifically? Should tests pass without adding this but something failing?

Reply:
I was expecting the tests to pass without specifying `ssr.resolve.conditions: ['node', 'module-sync']`.
`playground/ssr-resolve/main.js` is bundled to `playground/ssr-resolve/dist/main.js` and that file has `import moduleSync from '@vitejs/test-module-sync'`. Since `@vitejs/test-module-sync` is not processed by Vite / Vitest, I expected the resolver to use `module-sync` condition (as it's included in `ssr.resolve.externalConditions`). But it used the `ssr.resolve.conditions` and that does not include `module-sync` condition. So the resolution fails (`pkg-module-sync` only has `module-sync` condition).

Reply:
No ideal but this is how it works currently. ``import(`${testDir}/dist/main.mjs`)`` is transformed and evaluated through vite-node and each package's external check is done after resolved through `resolve.conditions` and it's not a same behavior as Vite SSR external. So, from what I know, this is not possible.

> Is it possible to configure Vitest to only use module-sync for externalized files?

Reply:
I'm not entirely sure the purpose of this test. Is this to confirm Vitest's behavior (with current PR change) works same as Vite SSR? Or do you want to test importing `dist/main.js` on a pure node runtime?

Reply:
Vitest has its own externalization logic because Vite only supports externalizing in SSR, but we need the same behavior for JSDOM. We resolve the ID with `pluginContainer.resolveId` and check if it needs to be external